src/internal/RSocketConnectionManager.cpp

```diff
@@ -87,8 +88,20 @@ void RSocketConnectionManager::manageConnection(
       *this, socket, eventBase);
   newNetworkStats->inner = std::move(socket->networkStats());
   socket->networkStats() = std::move(newNetworkStats);
+  sockets_.lock()->insert({std::move(socket), {eventBase, resumeToken}});
```

Comment:
The ConnectionManager stores shared_ptrs to all RSocketStateMachines.  It later uses it to close/destroy the RSocketStateMachines properly on exit (whenever RSocketServer/RSocketClient are destroyed).  We anyway store all the StateMachines. I don't think adding a pointer to these stateMachines would affect scaling signifcantly.  We do acquire lock before adding/removing the StateMachines, but I don't think it will be in the critical path.